WordPress/Classes/Services/MediaService.m

```diff
@@ -586,6 +654,14 @@ - (void)imageForMedia:(Media *)mediaInRandomContext
         }
         WPImageSource *imageSource = [WPImageSource sharedSource];
         void (^successBlock)(UIImage *) = ^(UIImage *image) {
+            // Check the media hasn't been deleted whilst we were loading.
+            if (!media || media.isDeleted) {
+                if (success) {
+                    success(nil);
```

Comment:
I've adjusted this to return a new, empty `UIImage`. Alternative would be to change the method signature, but I think it's probably best to leave that as-is?